docs/content/guides/dagster/local-to-production-tutorial.mdx

```diff
@@ -0,0 +1,448 @@
+- We need to be confident that data won't be accidentally overwritten (for example because a user forgot to change a configuration value).
+
+Using a few Dagster concepts, we can easily tackle this task! If you’re new to Dagster, here’s an overview of the main concepts we’ll be using in this guide (if you’re already familiar with Dagster, feel free to skip to the next paragraph):
+
+- [Asset](/concepts/software-defined-assets) - an asset is a software object that models a data asset. The prototypical example is a table in a database or a file in cloud storage.
+- [Resource](/concepts/resource) - a resource is an object that models a connection to a (typically) external service. Resources can be shared between assets, and different implementations of resources can be used depending on environment. For example, a resource may provide methods to send messages in Slack.
+- [IO manager](/concepts/io-management/io-managers) - an IO manager is a special kind of resource that handles storing and loading assets. For example, if we wanted to store assets in S3, we would use Dagster’s built in S3 IO manager.
+- [Run config](/concepts/configuration/config-schema) - assets and resources sometimes require configuration to set certain values (like the password to a database). Run config allows you to set these values at run time. In this guide, we will also use an API to set some default run configuration.
+
+Using these Dagster concepts we will:
+
+- Write three assets: the full Hacker News dataset, data about comments, and data about stories.
+- Use Dagster's Snowflake IO manager to store the datasets in [Snowflake](https://www.snowflake.com/).
+- Set up our Dagster [repository](/concepts/repositories-workspaces/repositories) so that the configuration for the Snowflake IO manager is automatically supplied based on the environment where the code is running.
+
+<Note>
+  At the end of this guide, you'll find an optional section that demonstrates
+  writing a resource to fetch data from Hacker News and a pattern for mocking
+  resources that can be helpful when writing unit tests.
+</Note>
+
+### Setup
+
+If you want to follow along with this guide you have two options:
+
+#### Clone the example code:
+
+This will give you the final version of the code we will write in this guide
+
+```bash
+pip install dagster, dagit
+dagster new-project --example local_to_prod
+cd local_to_prod
+pip install .
+```
+
+#### Start with a scaffolded project:
+
+This will create an empty project where you can copy the code as we write it in this guide
+
+```bash
+pip install dagster, dagit
+dagster new-project --quick-start local_to_prod
+cd local_to_prod
+pip install pandas pytest dagster-snowflake dagster-snowflake-pandas
+```
+
+### Part one: Local development
+
+In this section we will:
+
+- Write our assets.
+- Add the assets to our repository.
+- Add run configuration for the Snowflake IO manager.
+- Materialize assets in Dagit.
```

Comment:
```suggestion
- Write our assets
- Add the assets to our repository
- Add run configuration for the Snowflake I/O manager
- Materialize assets in Dagit
```

[2]

Our docs don't typically include periods at the end of bulleted list items unless there's more than one complete sentence.